packages/core/src/test/codewhisperer/commands/transformByQ.test.ts

```diff
@@ -297,3 +297,43 @@ describe('transformByQ', function () {
         assert.deepStrictEqual(actual, expected)
     })
 })
+
+describe('resetDebugArtifacts', () => {
+    it('should remove the directory containing the pre-build log file if it exists', async () => {
+        const dirPath = await createTestWorkspaceFolder()
+        const preBuildLogFilePath = path.join(dirPath.uri.fsPath, 'DummyLog.log')
+        await toFile('', preBuildLogFilePath)
+        transformByQState.setPreBuildLogFilePath(preBuildLogFilePath)
+
+        await resetDebugArtifacts()
+
+        assert.strictEqual(fs.existsSync(preBuildLogFilePath), false)
+        assert.strictEqual(transformByQState.getPreBuildLogFilePath(), '')
+    })
+
+    it.skip('should not remove any directory if the pre-build log file path is not set', async () => {
+        transformByQState.setPreBuildLogFilePath('')
+
+        const fsSpy = sinon.spy(fs, 'rmSync')
+
+        await resetDebugArtifacts()
+
+        sinon.assert.notCalled(fsSpy)
+        assert.strictEqual(transformByQState.getPreBuildLogFilePath(), '')
+        sinon.restore()
+    })
+
+    it.skip('should not remove any directory if the pre-build log file does not exist', async () => {
+        const preBuildLogFilePath = 'non/existent/path/to/pre-build.log'
+        transformByQState.setPreBuildLogFilePath(preBuildLogFilePath)
+
+        const fsSpy = sinon.spy(fs, 'rmSync')
```

Comment:
This is a brittle test. If gumby changes its implementation (e.g. to use [fsCommon](https://github.com/aws/aws-toolkit-vscode/blob/696823d36fa374928731976c1b7cd1eaba28da82/packages/core/src/srcShared/fs.ts#L32), as is strongly recommended) then these tests fail, even if the behavior is correct. Instead the test should check that the directory is actually deleted, without mocking fs functions.

Reply:
It's fair to not want tests to check whether specific unwrapped library calls are being made. 

Checking to see that the directory is actually deleted is the implementation of the test I've made (above this one) -- these two tests are checking that a non-existent directory isn't deleted, which is tricky when there's no side-effects to validate. 

After thinking about it, I decided to simplify this test, removing the mock